src/browser/BrowserService.cpp

```diff
@@ -219,6 +219,39 @@ QJsonObject BrowserService::getDatabaseGroups()
     return result;
 }
 
+QJsonArray BrowserService::getDatabaseEntries()
+{
+    auto db = getDatabase();
+    if (!db) {
+        return {};
+    }
+
+    Group* rootGroup = db->rootGroup();
+    if (!rootGroup) {
+        return {};
+    }
+
+    if (!browserSettings()->allowGetDatabaseEntriesRequest()) {
+        return {};
+    }
```

Comment:
This check should be in the `BrowserAction::handleGetDatabaseEntries` so you can issue an appropriate error "Access to all entries is denied"

Reply:
Done